packages/datadog-plugin-aws-sdk/test/bedrock.spec.js

```diff
@@ -0,0 +1,224 @@
+              prompt: prompt,
+              temperature: temperature,
+              p: topP,
+              k: topK,
+              max_tokens: maxTokens
+            },
+            response: {
+              id: '91c65da4-e2cd-4930-a4a9-f5c68c8a137c',
+              generations: [
+                {
+                  id: 'c040d384-ad9c-4d15-8c2f-f36fbfb0eb55',
+                  text: ' The capital of France is Paris. \n',
+                  finish_reason: 'COMPLETE'
+                }
+              ],
+              prompt: 'What is the capital of France?'
+            }
+
+          },
+          {
+            provider: META,
+            modelId: 'meta.llama3-70b-instruct-v1',
+            requestBody: {
+              prompt: prompt,
+              temperature: temperature,
+              top_p: topP,
+              max_gen_len: maxTokens
+            },
+            response: {
+              generation: '\n\nThe capital of France is Paris.',
+              prompt_token_count: 10,
+              generation_token_count: 7,
+              stop_reason: 'stop'
+            }
+          },
+          {
+            provider: MISTRAL,
+            modelId: 'mistral.mistral-7b-instruct-v0',
+            requestBody: {
+              prompt: prompt,
+              max_tokens: maxTokens,
+              temperature: temperature,
+              top_p: topP,
+              top_k: topK
+            },
+            response: {}
+          }
+        ]
+
+        models.forEach(model => {
+          it(`should invoke model for provider:${model.provider}`, done => {
+            const request = {
+              body: JSON.stringify(model.requestBody),
+              contentType: 'application/json',
+              accept: 'application/json',
+              modelId: model.modelId
+            }
+
+            const response = JSON.stringify(model.response)
+
+            nock('http://127.0.0.1:4566')
+              .post(`/model/${model.modelId}/invoke`)
+              .reply(200, response)
+
+            const command = new AWS.InvokeModelCommand(request)
+
+            // TODO mock the send command to return a specific response
+            bedrockRuntimeClient.send(command, (err) => {
+              if (err) return done(err)
+
+              agent.use(traces => {
+                const span = traces[0][0]
+                expect(span.meta).to.include({
+                  'aws.operation': 'invokeModel',
+                  'aws.bedrock.request.model': model.modelId.split('.')[1],
+                  'aws.bedrock.request.model_provider': `${model.provider}`,
+                  'aws.bedrock.request.prompt': prompt
+                })
+                expect(span.metrics).to.include({
+                  'aws.bedrock.request.temperature': temperature,
+                  'aws.bedrock.request.top_p': topP,
+                  'aws.bedrock.request.max_tokens': maxTokens
+                })
+              }).catch(done)
+              done()
+            })
```

Comment:
```suggestion
            agent.use(traces => {
              const span = traces[0][0]
              expect(span.meta).to.include({
                'aws.operation': 'invokeModel',
                'aws.bedrock.request.model': model.modelId.split('.')[1],
                'aws.bedrock.request.model_provider': `${model.provider}`,
                'aws.bedrock.request.prompt': prompt
              })
              expect(span.metrics).to.include({
                'aws.bedrock.request.temperature': temperature,
                'aws.bedrock.request.top_p': topP,
                'aws.bedrock.request.max_tokens': maxTokens
              })
            }).then(done, done)
            
            bedrockRuntimeClient.send(command, (err) => {
              if (err) return done(err)
            })
```
I would change it to something like this. the way it was before, having `done()` at the end of the bedrock callback would just end the test and pass it before the callback in `agent.use` executed. This way, the test will only end when either the `send` command fails, the `agent.use` callback fails, or the `agent.use` callback succeeds. Tried it out locally and it looks like a couple tests were silently failing (AI21 and Anthropic).